scripts/prepare_ports.py

```diff
@@ -31,12 +32,43 @@ def download_ports():
     return wpi_csv
 
 
+def filter_ports(dataframe):
+    """
+    Filters ports based on their harbor size and returns a DataFrame containing
+    only the largest port for each country.
+    """
+    # Filter large sized ports
+    large_ports = dataframe[dataframe["Harbor Size"] == "Large"]
+    countries_with_large_ports = large_ports["country"].unique()
+
+    # Filter out countries with large ports
+    remaining_ports = dataframe[~dataframe["country"].isin(countries_with_large_ports)]
+
+    # Filter medium sized ports from remaining ports
+    medium_ports = remaining_ports[remaining_ports["Harbor Size"] == "Medium"]
+    countries_with_medium_ports = medium_ports["country"].unique()
+
+    # Filter out countries with medium ports
+    remaining_ports = remaining_ports[
+        ~remaining_ports["country"].isin(countries_with_medium_ports)
+    ]
+
+    # Filter small sized ports from remaining ports
+    small_ports = remaining_ports[remaining_ports["Harbor Size"] == "Small"]
+
+    # Combine all filtered ports
+    filtered_ports = pd.concat([large_ports, medium_ports, small_ports])
+
+    return filtered_ports
+
+
 if __name__ == "__main__":
     if "snakemake" not in globals():
         from _helpers import mock_snakemake
 
         snakemake = mock_snakemake("prepare_ports")
 
+    config = snakemake.config
```

Comment:
Is this needed? if not better removed.
Parameters shall be passed using "params" key in snakemake so that snakemake can better handle re-triggering when appropriate.